api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenOnARTandFormulationsDataset.java

```diff
@@ -1,24 +1,784 @@
+            + "                SELECT p.patient_id , cn.name AS result_Value "
+            + "                FROM patient p "
+            + "                         INNER JOIN patient_program pp ON p.patient_id = pp.patient_id "
+            + "                         INNER JOIN program pgr ON pp.program_id = pgr.program_id "
+            + "                         INNER JOIN patient_state ps on pp.patient_program_id = ps.patient_program_id "
+            + "                         INNER JOIN program_workflow_state pws  on ps.state = pws.program_workflow_state_id          "
+            + "                         INNER JOIN concept_name cn  on pws.concept_id = cn.concept_id          "
+            + "                WHERE p.voided = 0 "
+            + "                  AND pp.voided = 0 "
+            + "                  AND ps.voided = 0 "
+            + "                  AND ps.state = ${6269} "
+            + "                  AND pgr.program_id = ${5} "
+            + "                  AND cn.locale = 'pt' "
+            + "                  AND ps.start_date >= DATE_SUB(:endDate, INTERVAL 210 DAY) "
+            + "                  AND ps.end_date <= :endDate "
+            + "               UNION  "
+            + "                SELECT p.patient_id, o.value_coded  AS result_Value "
+            + "                FROM patient p "
+            + "                         INNER JOIN encounter e on p.patient_id = e.patient_id "
+            + "                         INNER JOIN obs o on e.encounter_id = o.encounter_id "
+            + "                WHERE p.voided = 0 "
+            + "                  AND e.voided = 0 "
+            + "                  AND o.voided = 0 "
+            + "                  AND e.encounter_type = ${6} "
+            + "                  AND e.location_id = :location "
+            + "                  AND o.concept_id = ${23761} "
+            + "                  AND o.value_coded = ${1065} "
+            + "                  AND e.encounter_datetime "
+            + "                    BETWEEN DATE_SUB( :endDate, INTERVAL 210 DAY ) AND :endDate "
+            + ") AS final_query";
+
+    StringSubstitutor substitutor = new StringSubstitutor(valuesMap);
+
+    spdd.setQuery(substitutor.replace(sql));
+    return spdd;
+  }
+
+  /**
+   * 8
+   *
+   * @return
+   */
+  private DataDefinition getLastARVRegimen() {
+    SqlPatientDataDefinition spdd = new SqlPatientDataDefinition();
+    spdd.setName("Last ARV Regimen (FILA)");
+    spdd.addParameter(new Parameter("location", "location", Location.class));
+    spdd.addParameter(new Parameter("endDate", "endDate", Date.class));
+
+    Map<String, Integer> valuesMap = new HashMap<>();
+    valuesMap.put("18", hivMetadata.getARVPharmaciaEncounterType().getEncounterTypeId());
+    valuesMap.put("1088", hivMetadata.getRegimeConcept().getConceptId());
+
+    String sql =
+        " SELECT p.patient_id, ob.value_coded "
```

Comment:
Maybe we can add a join to concept_name and use the concept_id to return the Regimen name

Reply:
done